--- a/pkgs/sops-install-secrets/main.go
+++ b/pkgs/sops-install-secrets/main.go
@@ -239,9 +239,7 @@ func createSymlink(targetFile string, path string, owner int, group int, userMod
 			if os.IsNotExist(err) {
 				continue
 			} else if err != nil {
 				return fmt.Errorf("cannot read symlink '%s': %w", path, err)
-			} else if linksAreEqual(linkTarget, targetFile, stat, owner, group) {
-				return nil
 			}
 		}
 		if err := os.Remove(path); err != nil {
