--- a/pkgs/sops-install-secrets/main.go
+++ b/pkgs/sops-install-secrets/main.go
@@ -235,13 +235,11 @@
 			return fmt.Errorf("cannot stat '%s': %w", path, err)
 		}
 		if stat.Mode()&os.ModeSymlink == os.ModeSymlink {
-			linkTarget, err := os.Readlink(path)
+			_, err := os.Readlink(path)
 			if os.IsNotExist(err) {
 				continue
 			} else if err != nil {
 				return fmt.Errorf("cannot read symlink '%s': %w", path, err)
-			} else if linksAreEqual(linkTarget, targetFile, stat, owner, group) {
-				return nil
 			}
 		}
 		if err := os.Remove(path); err != nil {
